src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -614,142 +615,197 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
+         return(value)
+      if (tolower(value) %in% c("y", "yes", "on"))
+         return(TRUE)
+      if (tolower(value) %in% c("n", "no", "off"))
+         return(FALSE)
+      else
+         return(value)
+   })
+   
+   opts
 })
 
-.rs.addFunction("evaluateChunkOptions", function(code)
+.rs.addFunction("parseYamlOpt", function(opt, commentPrefix = "#")
 {
-  opts <- list()
-
-  Encoding(code) <- "UTF-8"
-  # if several lines of code are passed, we expect the first line to contain the standard Rmd chunk options
-  # the next several lines may or may not contain YAML-style chunk options
-  code <- unlist(strsplit(code, "\n", fixed = TRUE))
-  rmdChunkOpts <- code[[1]]
-  yamlChunkOpts <- c()
-
-  for (line in code[-1]) {
-    match <- unlist(regmatches(line, regexec(.rs.reYamlOptChunkBegin(), line)))
-    # there may be variable number of yaml chunk opts, but they should all be at the top of the chunk
-    if (length(match) == 0)
-        break
-    yamlChunkOpts <- c(yamlChunkOpts, match)
-  }
-  
-  if (length(yamlChunkOpts) > 0)
-  {
-     opts <- tryCatch(
-     .rs.parseYamlOpt(yamlChunkOpts),
-     error = function(e) {
-        warning("Failed to parse chunk options in body:\n", e)
-     })
-  }
-
-  # strip chunk indicators if present
-  matches <- unlist(regmatches(rmdChunkOpts, regexec(.rs.reRmdChunkBegin(), rmdChunkOpts)))
-  if (length(matches) > 1)
-    rmdChunkOpts <- matches[[2]]
-
-  tryCatch({
-    # if this is the setup chunk, it's not included by default
-    setupIndicator <- "r setup"
-    if (identical(substring(rmdChunkOpts, 1, nchar(setupIndicator)),
-                  setupIndicator)) {
-      opts$include <- FALSE
-    }
-
-    # extract and remove engine name (can be overridden below with engine=)
-    opts$engine <- unlist(strsplit(rmdChunkOpts, split = "(\\s|,)+"))[[1]]
-    rmdChunkOpts <- substring(rmdChunkOpts, nchar(opts$engine) + 1)
-
-    # parse them, then merge with the defaults (evaluate in global environment)
-    opts <- .rs.mergeLists(opts,
-                           eval(substitute(knitr:::parse_params(rmdChunkOpts)),
-                                envir = .GlobalEnv))
-
-    # if a chunk option is provided in both body (yaml chunk options) and header (rmdChunkOpts)
-    # the body should override the header option
-    opts <- opts[unique(names(opts))]
-    
-    # convert T, F for R code chunk options to TRUE, FALSE as appropriate
-    opts <- lapply(opts, function(opt) {
-      if (identical(opt, as.name("T")))
-        TRUE
-      else if (identical(opt, as.name("F")))
-        FALSE
-      else
-        opt
-    })
+   # remove comment prefix
+   commentPrefixPattern <- sprintf("^%s\\|\\s*", commentPrefix)
+   opt <- sub(commentPrefixPattern, "", opt)
+   
+   # check for YAML entry, mimicing code in knitr
+   if (grepl("^[^ :]+:($|\\s)", opt[[1]])) {
+      meta <- .rs.tryCatch(.rs.parseYamlOptImpl(opt))
+      if (!inherits(meta, "error"))
+         return(meta)
```

Comment:
So if it matches the regex but then fails to parse with `.rs.parseYamlOptImpl`, it will fall back to rmd style chunk params? does that make sense?

Reply:
Hmm, maybe not... here's what `knitr` does:

https://github.com/yihui/knitr/blob/42f6b3ad75f8245d26c1de3d3e0b9fa018566af3/R/parser.R#L301-L312

So we should probably just fail in a similar way.